source/fe/mapping_fe.cc

```diff
@@ -2142,17 +2141,21 @@ MappingFE<dim, spacedim>::compute_mapping_support_points(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell) const
 {
   // get the vertices first
-  std::vector<Point<spacedim>> a(cell->n_vertices());
+  std::vector<Point<spacedim>> vertices(cell->n_vertices());
 
   for (const unsigned int i : cell->vertex_indices())
-    a[i] = cell->vertex(i);
+    vertices[i] = cell->vertex(i);
 
-  if (this->polynomial_degree > 1)
-    {
-      Assert(false, ExcNotImplemented());
-    }
+  std::vector<Point<spacedim>> mapping_support_points(
+    fe->get_unit_support_points().size());
+
+  Table<2, double> weights;
+
+  cell->get_manifold().get_new_points(vertices,
+                                      weights,
+                                      mapping_support_points);
 
-  return a;
+  return mapping_support_points;
```

Comment:
@luca-heltai I think this matches what you described. But what I am not sure about how to integrate line and quad manifolds here, like it is done here:

https://github.com/dealii/dealii/blob/9f9205129e30e174594e3103136c5d8f39a2c635/source/fe/mapping_q_generic.cc#L1544-L1639

Reply:
For the moment, put an assertion that is thrown if the manifold ids of the subobjects are different than the one of the cell. What you wrote works for objects where everybody has the same manifold id. 

The idea would be to loop over all subobjects, querying (if any) all the support points that live on the edges, quads/tri, hex/tet/etc first, and then using the same thing you did here with the vertices of the subobjects and the support points that live in the middle of the subobjects.

Reply:
I have added the code. What would be an appropriate manifold to test with?

Reply:
Hyper shell, or cylinder_shell. They would work with a polar manifold, or a cylindrical manifold.